src/main/java/org/jboss/pnc/konfluxtooling/logging/UploadLogCommand.java

```diff
@@ -0,0 +1,89 @@
+package org.jboss.pnc.konfluxtooling.logging;
+
+import java.io.File;
+import java.math.BigInteger;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.MessageDigest;
+import java.time.OffsetDateTime;
+
+import org.jboss.pnc.bifrost.upload.BifrostLogUploader;
+import org.jboss.pnc.bifrost.upload.LogMetadata;
+import org.jboss.pnc.bifrost.upload.TagOption;
+
+import io.quarkus.logging.Log;
+import picocli.CommandLine;
+
+@CommandLine.Command(name = "upload-log")
+public class UploadLogCommand implements Runnable {
+
+    private static final int DEFAULT_MAX_RETRIES = 4;
+
+    private static final int DEFAULT_DELAY_SECONDS = 60;
+
+    @CommandLine.Option(names = "--file", required = true)
+    String logFile;
+
+    @CommandLine.Option(names = "--bifrost-url", required = true)
+    String bifrostURL;
+
+    @CommandLine.Option(names = "--max-retries")
+    int maxRetries = DEFAULT_MAX_RETRIES;
+
+    @CommandLine.Option(names = "--delay-seconds")
+    int delaySeconds = DEFAULT_DELAY_SECONDS;
+
+    @CommandLine.Option(names = "--process-context", required = true)
+    String processContext;
+
+    @CommandLine.Option(names = "--process-context-variant", required = true)
+    String processContextVariant;
+
+    @CommandLine.Option(names = "--tmp", required = true)
+    String tmp;
+
+    @CommandLine.Option(names = "--request-context", required = true)
+    String requestContext;
```

Comment:
If I understand correctly this is uploading the single `build.log` that is saved in https://github.com/redhat-appstudio/jvm-build-service/pull/2327 ? 

So IMHO I think it would be better if the 'final upload' of both jars (etc) and the build.log is handled in the same command (i.e. https://github.com/project-ncl/konflux-tooling/blob/main/src/main/java/org/jboss/pnc/konfluxtooling/deploy/DeployCommand.java ). 

I don't think the bifrost parameters should be required ; instead if they are not specified then no log upload should be performed?

Also did the PNC code have any tests to be ported across with it?

Reply:
I would go for a generic uploader that does not care about the content and can be used in any step, like wget or curl.

Reply:
@matejonnet I guess this code for uploading log is a bit specific for PNC. I don't know the details but looking at the those metadata properties would reveal that. I'd suggest we stick to it. 
@rnc I updated the code adding comments for those params. I don't know, at the moment, which one is required so I removed all 'required = true'. Also, as you suggested, skip it if bifrostURL is blank.

Reply:
@rnc In my viewpoint, it is better to separate the maven deployment with log upload. Although this means a new step in pipeline, it is easy to understand and maintain. As you can see, uploading log introduces a bunch of arguments. If we mix them to one command, it won't be easy to tell which is for which when people look at the pipeline args.

Reply:
I agree with both statements, keep the existing tool for log upload and keep the separate steps.

Reply:
@matejonnet I'm not convinced we need a generic tool - we already have the maven upload tooling but that is using the maven deployment code so isn't applicable.  This tool is highly specific to PNC so  how would the extra parameters be passed?
Further, I'm not convinced about keeping it separate - while it means the task parameters will be mixed it reduces the task step cpu/ memory cost (and bundle size for that matter) - I'm already seeing issues with running the buildah-oci task in some environments so I think we should reconsider this approach. Invoking as part of the deploy step will reduce the overall impact IMHO.

Reply:
Regarding cpu/memory, my personal experience is that it is a question of quota, something to be argued at later stage. Also, I see the current deployment step limits the cpu/mem to a minimal level. Maybe we don't need to worry too much about it at this moment?

Reply:
I'm sorry but I don't see the need to allocate another step with its own image initialisation/cpu/memory/processing time/etc giving we're only uploading a single file and could easily be rolled into the existing step. I think this is something we should all discuss in the new year.